server/index.js

```diff
@@ -29,6 +29,24 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const sql = `
+    select *
+      from "products"
+     where "productId" = $1;
+  `;
+  const value = [req.params.productId];
+  db.query(sql, value)
+    .then(result => {
+      if (result.rows[0]) {
+        res.json(result.rows[0]);
+      } else {
+        next(new ClientError(`cannot ${req.method} ${req.originalUrl}`, 404));
```

Comment:
This error message could be more specific and tell the client that the product doesn't exist.